File: 프로그래머스/1/77884. 약수의 개수와 덧셈/약수의 개수와 덧셈.cpp
#include <string>
#include <vector>
#include <iostream>

using namespace std;

int solution(int left, int right) {
    int answer = 0;

    int cnt = 0;
    for(int i = left; i<= right; ++i)
    {
        for(int j = 1; j <= i / 2; ++j)
        {
            if(i % j == 0)
                ++cnt;
        }
        cnt += 1;
        if(cnt % 2 == 0)
            answer += i;
        else
            answer -= i;
        cnt = 0;
    }
    
    return answer;
}

File: 프로그래머스/1/77884. 약수의 개수와 덧셈/약수의 개수와 덧셈.cs
using System;

public class Solution {
    public int getDivisor(int num)
    {
        int totalCnt = 0;
        for(int i = 1; i <= num; ++i)
        {
            if(num % i == 0)
                ++totalCnt;
        }
        return totalCnt;
    }
    
    public int solution(int left, int right) {
        int answer = 0;
        
        for(int i = left; i <= right; ++i)
        {
            int divisorCnt = getDivisor(i);
            if(divisorCnt % 2 == 0)
                answer += i;
            else
                answer -= i;
        }
        
        
        return answer;
    }
}